test/testsimplifytypedef.cpp

```diff
@@ -2422,6 +2423,19 @@ class TestSimplifyTypedef : public TestFixture {
         ASSERT_EQUALS(expected, tok(code));
     }
 
+    void simplifyTypedef115() {     // ticket #6998
+        const char code[] = "typedef unsigned unsignedTypedef;\n"
+                            "unsignedTypedef t1 ;\n"
+                            "unsigned t2 ;";
+        const char expected[] = "unsigned int t1 ; "
+                                "unsigned int t2 ;";
+        // Tokenizer::simplifyStdType turns "unsigned t" into "int t"...
+        const char received[] = "int t1 ; "
+                                "int t2 ;";
+        TODO_ASSERT_EQUALS(expected, received, tok(code));
```

Comment:
Why TODO? unsigned is not lost, it is just removed as a token and stored in the isUnsigned attribute of the "int" token. You would see this by calling tok(code, false) for example, as attributes are shown then.